controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Maintainer.java

```diff
@@ -91,4 +91,12 @@ public final String toString() {
         return name();
     }
 
+    private static long staggeredDelay(Controller controller, Duration interval) {
```

Comment:
Some of the things this code tries to avoid should be explicitly verified, like if `interval` is not positive, throw `IllegalArgumentException`...

If we can assume that `cluster.size() > 0` and `interval.toMillis() > 0`, then this can be simplified to
```
private static long staggeredDelay(Controller controller, Duration interval) {
    List<HostName> cluster = controller.curator().cluster();
    long offset = interval.toMillis() * cluster.indexOf(controller.hostname()) / cluster.size();
    return Math.floorMod(offset - controller.clock().millis(), interval.toMillis());
}
```

This has the added benefit of starting within `interval / cluster.size()`, instead of waiting until next interval start.

Reply:
Yes, you're absolutely right. Hang on!

Reply:
I want to wait at least `interval.toMillis() / cluster.size()` until running the job — the previous wait was always `interval.toMillis()`, and probably for some reason.

Reply:
Yes, to avoid maintainers running during unit tests ... :)

Reply:
This does not look right... A test would definitely help. I refactored your method a little bit:
```
static long staggeredDelay(List<HostName> cluster, HostName hostName, Instant now, Duration interval) {
    if ( ! cluster.contains(hostName))
        return interval.toMillis();

    long timeUntilNextRun = Math.floorMod(-now.toEpochMilli(), interval.toMillis() / cluster.size());
    return timeUntilNextRun + (1 + cluster.indexOf(hostName)) * interval.toMillis() / cluster.size();
}
```
Testing this with:
```
List<HostName> hostNames = Stream.of("cfg1", "cfg2", "cfg3")
        .map(HostName::from)
        .collect(Collectors.toList());

Instant now = Instant.ofEpochMilli(324232003);
for (HostName hostName : hostNames) {
    System.out.println(Maintainer.staggeredDelay(hostNames, hostName, now, Duration.ofSeconds(1)));
}
```
Prints
```
445
778
1112
```
They should all be > 1000...
Adding `interval` to the final result in my implementation above:
```
static long staggeredDelay(List<HostName> cluster, HostName hostName, Instant now, Duration interval) {
    if ( ! cluster.contains(hostName))
        return interval.toMillis();

    long offset = interval.toMillis() * cluster.indexOf(hostName) / cluster.size();
    return Math.floorMod(offset - now.toEpochMilli(), interval.toMillis()) + interval.toMillis();
}
```
Gives:
```
1003
1336
1669
```

Reply:
No, the first one is correct. `interval / cluster-size` is enough wait, but `0` is not :p